src/components/share/ShareDirective.js

```diff
@@ -69,13 +81,47 @@
                 }).success(function(response) {
                   scope.permalinkValue = response.shorturl;
                   scope.urlShortened = true;
+                  scope.$applyAsync(function() {
+                    // Auto-select the shortened permalink (not on mobiles)
+                    if (!gaBrowserSniffer.mobile) {
+                      $('#permalinkInput')[0].setSelectionRange(0, 9999);
+                    }
+                  });
                 });
               };
 
+              // Use clipboard API to copy URL in OS clipboard
+              scope.copyUrl = function(type) {
+                // Select the permalink anchor text
+                if (type == 'permalink') {
+                  var inputElement = $('#permalinkInput');
+                } else if (type == 'embed') {
+                   var inputElement = $('.ga-embed-input');
+                }
```

Comment:
this `if` doesn't look very generic to me. Why not create a `ga-copy`  directive which takes a  jquery selector as parameter : 

`<button ga-copy=".ga-embed-input"></button>`